src/relax/ir/dataflow_matcher.cc

```diff
@@ -540,33 +541,48 @@ struct RNode {
 /**
  * \brief This method try to match a real node and a pattern node along with its neighbors.
  */
-static bool try_match(PNode* p, RNode* r, DFPatternMatcher* m,
-                      const std::map<const VarNode*, std::vector<const VarNode*>>& def2use,
-                      const std::map<const VarNode*, std::vector<const VarNode*>>& use2def) {
-  if (p->matched != nullptr && p->matched == r->ptr) return true;  // matched before.
-  if (!m->Match(GetRef<DFPattern>(p->ptr), GetRef<Var>(r->ptr))) return false;
+using UndoStack = std::stack<std::pair<PNode*, RNode*>>;
```

Comment:
Consider `std::stack<std::pair<PNode*, RNode*>, std::vector<std::pair<PNode*, RNode*>>>` for cache-friendliness.

Reply:
If we have perf consideration, we can do `std::vector` and do `std::copy` in L580.

Reply:
ok it seems it doesn't have to be a stack. I'll update.